src/services/milestones/sendNotification.js

```diff
@@ -126,7 +128,9 @@ const sendNotification = () => async context => {
             recipient: campaign.owner.email,
             user: campaign.owner.name,
             milestoneTitle: data.title,
+            milestoneId: data.milestoneId,
             campaignTitle: campaign.title,
+            campaignId: data.campaignId,
             amount: data.maxAmount,
             token: data.token,
           });
```

Comment:
Instead of `data` use `result`. If it's not clear you can take a look at this
https://docs.feathersjs.com/guides/basics/hooks.html#hook-context
`data` is data sent by user which doesn't have all milestone data necessarily, but `result` has. We can expect `data` to have all data just in `create` case.
I suggest using result for other calling `Notifications.milestoneProposed` too.